sql/core/src/main/scala/org/apache/spark/sql/execution/systemcatalog/InformationSchema.scala

```diff
@@ -0,0 +1,336 @@
+      // We should generalize this later.
+      sparkSession.sessionState.catalog.createTempView(s"$INFORMATION_SCHEMA.$name",
+        plan, overrideIfExists = true)
+    }
+  }
+
+  /**
+   * Compile filter array into single string condition.
+   */
+  private[systemcatalog] def getConditionExpressionString(filters: Array[Filter]): String = {
+    val str = filters.flatMap(InformationSchema.compileFilter).map(p => s"($p)").mkString(" AND ")
+    if (str.length == 0) "TRUE" else str
+  }
+
+  /**
+   * Convert filter into string expression.
+   */
+  private[systemcatalog] def compileFilter(f: Filter): Option[String] = {
+    Option(f match {
+      case EqualTo(attr, value) => s"$attr = ${compileValue(value)}"
+      case EqualNullSafe(attr, value) =>
+        s"(NOT ($attr != ${compileValue(value)} OR $attr IS NULL OR " +
+          s"${compileValue(value)} IS NULL) OR " +
+          s"($attr IS NULL AND ${compileValue(value)} IS NULL))"
+      case LessThan(attr, value) => s"$attr < ${compileValue(value)}"
+      case GreaterThan(attr, value) => s"$attr > ${compileValue(value)}"
+      case LessThanOrEqual(attr, value) => s"$attr <= ${compileValue(value)}"
+      case GreaterThanOrEqual(attr, value) => s"$attr >= ${compileValue(value)}"
+      case IsNull(attr) => s"$attr IS NULL"
+      case IsNotNull(attr) => s"$attr IS NOT NULL"
+      case StringStartsWith(attr, value) => s"${attr} LIKE '${value}%'"
+      case StringEndsWith(attr, value) => s"${attr} LIKE '%${value}'"
+      case StringContains(attr, value) => s"${attr} LIKE '%${value}%'"
+      case In(attr, value) => s"$attr IN (${compileValue(value)})"
+      case Not(f) => compileFilter(f).map(p => s"(NOT ($p))").orNull
+      case Or(f1, f2) =>
+        val or = Seq(f1, f2).flatMap(compileFilter(_))
+        if (or.size == 2) {
+          or.map(p => s"($p)").mkString(" OR ")
+        } else {
+          null
+        }
+      case And(f1, f2) =>
+        val and = Seq(f1, f2).flatMap(compileFilter(_))
+        if (and.size == 2) {
+          and.map(p => s"($p)").mkString(" AND ")
+        } else {
+          null
+        }
+      case _ => null
+    })
+  }
+
+  /**
+   * Converts value into string expression.
+   */
+  private[systemcatalog] def compileValue(value: Any): Any = value match {
+    case s: String =>
+      val escapedValue = if (s == null) null else StringUtils.replace(s, "'", "''")
+      s"'$escapedValue'"
+    case t: Timestamp => "'" + t + "'"
+    case d: Date => "'" + d + "'"
+    case a: Array[Any] => a.map(compileValue).mkString(", ")
+    case _ => value
+  }
+}
+
+/**
+ * Provides DATABASES relation consisting all databases.
+ */
+private class DatabasesRelationProvider extends SchemaRelationProvider {
+  override def createRelation(
+      sqlContext: SQLContext,
+      parameters: Map[String, String],
+      schema: StructType): BaseRelation = {
+    val sparkSession = sqlContext.sparkSession
+
+    new BaseRelation with PrunedFilteredScan {
```

Comment:
Ur, actually, I used `PrunedFilteredScan` because I thought`CatalystScan` is annotated Experimental and no classes extends that trait. Is there any know troubles with `PrunedFilteredScan`?